### PR TITLE
bgp: T3945: Add route-map for aggregate-address

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

Add a new feature, use route-map for bgp aggregate-address
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T3945

## Component(s) name

bgp
## How to test
VyOS configuration:
```
set policy route-map FOO rule 10 action 'permit'
set policy route-map FOO rule 10 set community '111:222'
set protocols bgp address-family ipv4-unicast aggregate-address 203.0.113.0/24 route-map 'FOO'
set protocols bgp local-as '65001'
commit
```
Expected FRR configuration:
```
!
router bgp 65001
 no bgp ebgp-requires-policy
 no bgp default ipv4-unicast
 no bgp network import-check
 !
 address-family ipv4 unicast
  aggregate-address 203.0.113.0/24 route-map FOO
 exit-address-family
!

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: data/templates/frr/bgpd.frr.tmpl [hunk 1]
```diff
@@ -266,6 +266,9 @@ router bgp {{ local_as }} {{ 'vrf ' ~ vrf if vrf is defined and vrf is not none
 {%     if afi_config.aggregate_address is defined and afi_config.aggregate_address is not none %}
 {%       for ip in afi_config.aggregate_address %}
   aggregate-address {{ ip }}{{ ' as-set' if afi_config.aggregate_address[ip].as_set is defined }}{{ ' summary-only' if afi_config.aggregate_address[ip].summary_only is defined }}
+{%         if afi_config.aggregate_address[ip].route_map is defined and afi_config.aggregate_address[ip].route_map is not none %}
```
Comment (dmbaturin) [inline data/templates/frr/bgpd.frr.tmpl:269]: I'm seeing a lot of that pattern: `if foo is defined and foo is not none`. I see where it's coming from, but I can't help but wonder: can we combine those into a single reusable macro? That could save us a lot of duplicate code and make templates more readable.

**Review by dmbaturin — APPROVED** [review]

File: data/templates/frr/bgpd.frr.tmpl (hunk 1 above)
Comment (c-po) [reply data/templates/frr/bgpd.frr.tmpl:269]: I am actually working on a custom filter but have not got it working so far.

Comment (dmbaturin) [reply data/templates/frr/bgpd.frr.tmpl:269]: Are there any fundamental obstacles, or it's just the Jinja2 API yak shaving that takes time?

Comment (c-po) [reply data/templates/frr/bgpd.frr.tmpl:269]: Its actually only gluing some Python code to Jinja2 - but testing consumes the most time.

**Comment by c-po** [comment]
Can you please also add IPv6 support @sever-sever?

**Comment by sever-sever** [comment]
> Can you please also add IPv6 support @sever-sever?

@c-po For 1.4 it's already included to "afi"
```
set protocols bgp address-family ipv6-unicast aggregate-address 2001:db8::/56 route-map FOO
```
frr:
```
 !
 address-family ipv6 unicast
  aggregate-address 2001:db8::/56 route-map FOO
 exit-address-family
!

```

**Pull request merged**